### PR TITLE
支持mesh子网格创建

### PR DESCRIPTION
**Pull request by wingcd**
支持mesh子网格创建

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
- it seems break compatibility
- could you please fix the eslint error?

**Comment by wingcd** [comment]
> * it seems break compatibility
> * could you please fix the eslint error?

好了

**Comment by minggo** [comment]
但是这个改动修改了接口，破坏了兼容性了。

**Comment by wingcd** [comment]
之前那个就有问题，只支持一个子网格，新版本可以直接用这个，实在不行可以新增一个接口？

**Comment by minggo** [comment]
新增接口比较好。

**Comment by minggo** [comment]
理论上 createDynamicMesh 是 createDynamicMeshes 的特例。那能用 createDynamicMeshes 实现 createDynamicMesh 吗？

**Comment by wingcd** [comment]
当然可以的

**Comment by minggo** [comment]
我的意思是类似这样的代码
```
static createDynamicMesh(...) {
    ...
    createDynamicMeshes(...);
}
```

**Comment by wingcd** [comment]
好的。可以，其实特例的参数就没用了而已

**Review by GengineJS — APPROVED** [review]

File: cocos/3d/misc/create-mesh.ts [hunk 1]
```diff
@@ -277,46 +277,54 @@ function getPadding (length: number, align: number): number {
 }
 
 function createDynamicMesh (primitiveIndex: number, geometry: IDynamicGeometry, out?: Mesh, options?: ICreateDynamicMeshOptions): Mesh {
-    options = options || { maxSubMeshes: 1, maxSubMeshVertices: 1024, maxSubMeshIndices: 1024 };
+    return createDynamicMeshes([geometry], out);
```
Comment (minggo) [inline cocos/3d/misc/create-mesh.ts:280]: 对 Options 的支持没有了？

**Comment by dumganhar** [comment]
@minggo This PR was automatically closed since the target branch was deleted. And the developer could not re-open this Pull Request. 
We need to check whether to cherry-pick the changes to 3.8.6.

**Comment by minggo** [comment]
This PR break compatibility, can no be merged.

**Pull request closed without merging**